ablySpec/RealtimeClientChannel.swift

```diff
@@ -526,6 +526,60 @@ class RealtimeClientChannel: QuickSpec {
 
                 }
 
+                // RTL6g
+                context("Identified clients with clientId") {
+
+                    // RTL6g1
+                    context("When publishing a Message with clientId set to null") {
+
+                        // RTL6g1a
+                        pending("should be unnecessary to set clientId of the Message before publishing") {
+                            let options = AblyTests.commonAppSetup()
+                            options.autoConnect = false
+                            let client = ARTRealtime(options: options)
+                            client.setTransportClass(TestProxyTransport.self)
+                            client.connect()
+                            defer { client.close() }
+
+                            let channel = client.channels.get("test")
+
+                            channel.publish("message", cb: { status in
+                                expect(status.state).to(equal(ARTState.Ok))
+                            })
+
+                            let transport = client.transport as! TestProxyTransport
+                            expect(transport.protocolMessagesSent.filter { $0.action == .Message }).toEventually(haveCount(1), timeout: testTimeout)
+
+                            let messageSent = transport.protocolMessagesSent.filter({ $0.action == .Message })[0]
+                            expect(messageSent.messages![0].clientId).to(beNil())
+                        }
+
+                        // RTL6g1b
+                        pending("should have clientId value as null for the Message when received") {
+                            let options = AblyTests.commonAppSetup()
+                            options.autoConnect = false
+                            let client = ARTRealtime(options: options)
+                            client.setTransportClass(TestProxyTransport.self)
+                            client.connect()
+                            defer { client.close() }
+
+                            let channel = client.channels.get("test")
+
+                            channel.publish("message", cb: { status in
+                                expect(status.state).to(equal(ARTState.Ok))
+                            })
+
+                            let transport = client.transport as! TestProxyTransport
+                            expect(transport.protocolMessagesReceived.filter { $0.action == .Message }).toEventually(haveCount(1), timeout: testTimeout)
+
+                            let messageReceived = transport.protocolMessagesReceived.filter({ $0.action == .Message })[0]
+                            expect(messageReceived.messages![0].clientId).to(beNil())
```

Comment:
I would also add a normal `subscribe` check to ensure that a message received using the normal documented API will deliver a message without a clientId to the listener